cura/Settings/MachineManager.py

```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
     #        Examples:
     #          - "my_profile - Fine" (only based on a default quality, no intent involved)
     #          - "my_profile - Engineering - Fine" (based on an intent)
-    @pyqtProperty("QVariantMap", notify = activeQualityDisplayNameChanged)
-    def activeQualityDisplayNameMap(self) -> Dict[str, str]:
-        global_stack = self._application.getGlobalContainerStack()
-        if global_stack is None:
-            return {"main": "",
-                    "suffix": ""}
+    @pyqtProperty("QList<QString>", notify = activeQualityDisplayNameChanged)
+    def activeQualityDisplayNameStringParts(self) -> [str]:
+        result_map = self.activeQualityDisplayNameMap
+        string_parts = list()
+
+        if result_map["custom_profile"] is not None:
+            string_parts.append(result_map["custom_profile"])
 
-        display_name = global_stack.quality.getName()
+        if result_map["intent_category"] is not "default":
+            string_parts.append(f"""{result_map["intent_name"]} - {result_map["profile"]}""")
+        else:
+            string_parts.append(result_map["profile"])
 
-        intent_category = self.activeIntentCategory
-        if intent_category != "default":
-            intent_display_name = IntentCategoryModel.translation(intent_category,
-                                                                  "name",
-                                                                  intent_category.title())
-            display_name = "{intent_name} - {the_rest}".format(intent_name = intent_display_name,
-                                                               the_rest = display_name)
+        if result_map["layer_height"]:
+            string_parts.append(f"""{result_map["layer_height"]}mm""")
```

Comment:
```suggestion
            string_parts.append(f"{result_map["layer_height"]}mm")
```

Reply:
I tried this, but this gave me linting errors in pycharm